### PR TITLE
More robust way of allocating memory in NumberArrayFactory

### PR DESCRIPTION
**Pull request by tinwelint**
also, the AvailableMemoryCalculator#RUNTIME uses the correct OS bean
method for figuring out how much free memory there's available.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java [hunk 1]
```diff
@@ -121,48 +123,66 @@ public LongArray newLongArray( long length, long defaultValue )
...
+            }
+            throw error;
         }
 
         @Override
         public IntArray newIntArray( long length, int defaultValue )
         {
-            return mostAppropriateFactory( length, 4 ).newIntArray( length, defaultValue );
+            OutOfMemoryError error = null;
+            for ( NumberArrayFactory candidate : candidateFactories( length, 8 ) )
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java]: 8->4

**Comment by davidegrohmann** [comment]
@tinwelint maybe we should add some test for `newIntArray` since the current ones call only `newLongArray`.

**Comment by tinwelint** [comment]
Good catch, I added tests for int and fixed the problem.

**Comment by tinwelint** [comment]
It's quite annoying, but good I'd guess, that CI has very low memory machines that runs the tests... I'll reconsider how to do this

**Comment by tinwelint** [comment]
@davidegrohmann how about now? I made a change where NumberArrayFactory.Auto doesn't use AvailableMemoryCalculator at all, since it's quite likely to not be 100% true anyway. It'll instead try to just allocate and the system will have to throw OOM enough memory not available. Auto has a fixed list of candidates to try with, sort of like before.

This is the most robust way I've come up with as of yet.

**Comment by davidegrohmann** [comment]
@tinwelint it looks good to me.

**Pull request merged**